src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -543,6 +569,21 @@ def construct_payload(self):
             else:
                 set_managed_identity(self.cmd, self.get_argument_resource_group_name(), self.containerapp_def, user_assigned=[self.get_argument_registry_identity()])
 
+        if self.get_argument_source():
+            self.set_up_create_containerapp_if_source_or_repo()
+
+        if self.get_argument_repo():
+            container_app = self._get_containerapp_if_exists()
+            if container_app:
+                # Update the image to ensure that the container app is not re-created with mcr.microsoft.com/k8se/quickstart:latest
+                self.containerapp_def["properties"]["template"]["containers"][0]["image"] = container_app["properties"]["template"]["containers"][0]["image"]
```

Comment:
please check containers exist and count > 0 first

Reply:
The code on line 579 will be executed only if the Container App already exists, my understanding is that containers must exist if a Container App was already created. If `--image` was not provided by the user, it is defaulted to `mcr.microsoft.com/k8se/quickstart:latest`. Could you please confirm if there could be a case where containers does not exist?